ports/stm32/stm32_it.c

```diff
@@ -720,15 +720,15 @@ void USART6_IRQHandler(void) {
     IRQ_EXIT(USART6_IRQn);
 }
 
-#if defined(MICROPY_HW_UART7_TX)
+#if defined(UART8)
```

Comment:
Shouldn't it be UART7?

And, what is your rationale for changing this line from what it was?  (I would tend to agree with the change.)

Reply:
If I checked if UART7 was defined it would suggest that we support enabling UART7 if it's present without UART8 but that wouldn't work due to MICROPY_HW_MAX_UART being set to 6. e.g. on a board that has 7 UARTs

My rationale was that if MICROPY_HW_UART7_TX was defined users may expect UART7 to actually work if the pins for UART7 are defined - see above case regarding MICROPY_HW_MAX_UART.